src/ssz/tree_view.zig

```diff
@@ -176,6 +188,13 @@ pub fn TreeView(comptime ST: type) type {
                         &value,
                     ),
                 );
+                if (opt_old_node) |old_node| {
+                    // Multiple set() calls before commit() leave our previous temp nodes cached with refcount 0.
+                    // Tree-owned nodes already have a refcount, so skip unref in that case.
+                    if (old_node.value.getState(self.pool).getRefCount() == 0) {
+                        self.pool.unref(old_node.value);
+                    }
```

Comment:
why do we need the check for getRefCount() == 0 here?
Shouldn't the old node always be unrefed? (which will ref_count-- and possibly free the old node if ref_count gets set to 0)

Reply:
Oh yes, I am misundstanding the `commit` function.

Reply:
I think the `commit` function will help `unref` tree-owned node, we shouldn't call `unref`  here which this kind of node refcount should >0. But like below case, A is a new fresh node and ref count 0, but it replaced by B, it seems will leak because `commit` will not return it to the pool so we need  `unref` here with condition `old_node.value.getState(self.pool).getRefCount() == 0`.

```
setElement(A)
setElement(B)

commit()
```